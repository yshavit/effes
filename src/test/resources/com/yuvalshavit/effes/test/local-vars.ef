def foo True -> True:
  t1 = True
  t2 = t1
  print t2
  return t2

def main -> True: return foo True
