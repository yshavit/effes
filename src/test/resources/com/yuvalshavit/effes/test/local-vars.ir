def foo (True) -> True:
  AssignStatement
    var: t1 (pos=0)
    val:
      CtorInvoke
        type: True
  AssignStatement
    var: t2 (pos=1)
    val:
      VarExpression
        name: t1
        pos: 0
  MethodInvoke
    method:
      MethodInvoke
        method: print [built-in]
        arg0:
          VarExpression
            name: t2
            pos: 1
  ReturnStatement
    VarExpression
      name: t2
      pos: 1
