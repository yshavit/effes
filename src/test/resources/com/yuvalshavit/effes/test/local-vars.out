Your argument was: True
>> True
